### ISSUE TITLE
ArgumentCountError - Fixtures with DI and with DependentFixtureInterface

### ISSUE THREAD

**Issue by guillaume-a** [doctrine/data-fixtures#295]
Hello, 

I have a project with some fixtures for functionnal testing.
And one of my fixtures, for my User Entity, has a dependency injection

```
class UserFixtures extends Fixture {
	private $encoder;
		public function __construct(UserPasswordEncoderInterface $encoder)
	{
		$this->encoder = $encoder;
	}
	public function load( ObjectManager $manager ) {
		//.....
	}
}
```

Thanks to symfony 4 and autowiring, it works when I use command line.
Then, I wrote my functionnal test with the famous

`$this->loadFixtures(array('App\DataFixtures\UserFixtures'));`

At this point, I had an issue witch I solved with service_test.yaml

```
services:
    _defaults:
        public: true

    App\DataFixtures\UserFixtures:
        arguments: ['@security.password_encoder']
```

But now, I have more fixtures with dependencies.

```
class PermissionFixtures extends Fixture implements DependentFixtureInterface {
	public function load( ObjectManager $manager ) {
		//...
	}

	public function getDependencies()
	{
		return array(
			UserFixtures::class,
			EtablissementFixtures::class,
		);
	}
}
```

And I have this error : (the same I solved with injecting in services_test.yaml)

`ArgumentCountError : Too few arguments to function App\DataFixtures\UserFixtures::__construct(), 0 passed in /path/to/project/vendor/doctrine/data-fixtures/lib/Doctrine/Common/DataFixtures/Loader.php on line 193 and exactly 1 expected`

After some breakpoints, I found this in doctrine code : 

```
if ($fixture instanceof OrderedFixtureInterface) {
    $this->orderFixturesByNumber = true;
} elseif ($fixture instanceof DependentFixtureInterface) {
    $this->orderFixturesByDependencies = true;
    foreach($fixture->getDependencies() as $class) {
        if (class_exists($class)) {
            $this->addFixture($this->createFixture($class));
        }
    }
}
```

So, if my fixture implements DependentFixtureInterface it goes through createFixture method.

```
    protected function createFixture($class)
    {
        return new $class();
    }
```

And I see now, it just create the fixture with new UserFixture() and no argument is passed.

So my question is : 

Is this considered a bug ? 
Is there some kind of workaround ?

Thank you for reading.

**Comment by weaverryan** [comment]
Hi there!

Where does the $this->loadFixtures() method come from? I’m not familiar with it.

This metho appears to be circumventing the normal way that the fixtures are loaded. I can’t give more directions because I’m not at a computer, but the way you are loading the fixtures is at the root of the problem.

Cheers!

**Comment by guillaume-a** [comment]
Hi, 

`$this->loadFixtures()` comes from `Liip\FunctionalTestBundle\Test\WebTestCase`, the [FunctionalTestBundle](https://github.com/liip/LiipFunctionalTestBundle) I use for functionnal testing on symfony.

And they have a chapter on [fixtures](https://github.com/liip/LiipFunctionalTestBundle#database-tests) which I followed.

Regards.

**Comment by BonBonSlick** [comment]
@guillaume-a  tell me please how you solved that?

```
imports :
  - { resource: '../../parameters.xml' }

doctrine:
  dbal:
    default_connection: default
    connections       :
      default:
        driver :   pdo_sqlite
        path   :    '%kernel.cache_dir%/test.db'
        memory : false
        charset: UTF8
```

services test
```

  <services>
    <defaults autowire="true"
              autoconfigure="true"
              public="false"
    >
    </defaults>

    <service id="RoleFactoryInterface"
             alias="App\Domain\Role\Factory\RoleFactoryInterface"
             public="true"
    />
    <service id="UserFactoryInterface"
             alias="App\Domain\UserPack\User\Factory\UserFactoryInterface"
             public="true"
    />
  </services>

```
tests setUp()
```

    $loader        = new Loader();
    $container     = static::$container;
    $roleFactory                 = $container->get(RoleFactoryInterface::class);
    $roleFixtures                = new RoleFixtures($roleFactory);
    $roleFixtures->setContainer($container);
    $loader->addFixture($roleFixtures);
    $userFactory  = $container->get(UserFactoryInterface::class);
    $userFixtures = new UserFixtures($userFactory);
    $userFixtures->setContainer($container);
    $loader->addFixture($userFixtures);
```

ERROR
`ArgumentCountError: Too few arguments to function App\Infrastructure\Persistence\DataFixtures\RoleFixtures::__construct(), 0 passed in /var/www/hint/vendor/doctrine/data-fixtures/lib/Doctrine/Common/DataFixtures/Loader.php on line 193 and exactly 1 expected`

**Comment by alcaeus** [comment]
After taking another look, this looks like an issue not with this library but rather with the LiipFunctionalTestBundle not supporting dependency injection in fixtures which is provided using the `Doctrine\Bundle\FixturesBundle\Loader\SymfonyFixturesLoader` and compiler passes to ensure fixture services are injected into the fixture loader.

Relevant discussion can be fixed in https://github.com/liip/LiipFunctionalTestBundle/pull/432 and https://github.com/liip/LiipFunctionalTestBundle/issues/411. This will apparently be fixed in the upcoming 2.0 release. See https://github.com/liip/LiipFunctionalTestBundle/issues/381 for relevant discussion on this.

As this is not an issue with this library or our bundle, I'm closing this issue.

**Comment by BonBonSlick** [comment]
@alcaeus finally it worked with SymfonyFixturesLoader, thank you very much!
